pkg/sql/schemachanger/scplan/internal/rules/dep_index_and_column.go

```diff
@@ -85,22 +85,65 @@ func init() {
 					(*scpb.IndexName)(nil),
 					(*scpb.IndexPartitioning)(nil),
 					(*scpb.IndexComment)(nil),
+					(*scpb.IndexColumn)(nil),
 				),
 				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
 				targetStatusEq(from.target, to.target, scpb.ToPublic),
 				currentStatus(from.node, scpb.Status_BACKFILL_ONLY),
 				currentStatus(to.node, scpb.Status_PUBLIC),
 			}
 		})
+	registerDepRule(
+		"index existence precedes index dependents",
+		scgraph.SameStagePrecedence,
+		"index", "index-column",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type(
+					(*scpb.PrimaryIndex)(nil),
+					(*scpb.SecondaryIndex)(nil),
+				),
+				to.el.Type(
+					(*scpb.IndexColumn)(nil),
+				),
+				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
+				targetStatusEq(from.target, to.target, scpb.ToPublic),
+				currentStatus(from.node, scpb.Status_BACKFILL_ONLY),
+				currentStatus(to.node, scpb.Status_PUBLIC),
+			}
+		})
+	registerDepRule(
+		"temp index existence precedes index dependents",
+		scgraph.SameStagePrecedence,
+		"index", "index-column",
+		func(from, to nodeVars) rel.Clauses {
+			return rel.Clauses{
+				from.el.Type(
+					(*scpb.PrimaryIndex)(nil),
+					(*scpb.SecondaryIndex)(nil),
+				),
+				to.el.Type(
+					(*scpb.IndexColumn)(nil),
+				),
+				joinOnIndexID(from.el, to.el, "table-id", "index-id"),
+				targetStatus(from.target, scpb.Transient),
```

Comment:
Shouldn't this be `scpb.Transient_Absent`?

Reply:
I think it is what you mean : http://github.com/cockroachdb/cockroach/blob/8601bedf62a7e0534a99c33087cdafcc6d1015fd/pkg/sql/schemachanger/scpb/constants.go#L41-L41

though good call that it is confusing.